load-testing/load_test.py

```diff
@@ -0,0 +1,140 @@
+import argparse
+import json
+import multiprocessing
+import os
+import sys
+import time
+from datetime import datetime
+
+from config import (
+    DETECTOR_NAME,
+    DETECTOR_QUERY,
+    ENDPOINT_URL,
+    GROUNDLIGHT_API_TOKEN,
+    IMAGE_PATH,
+    LOG_FILE,
+    REQUESTS_PER_SECOND,
+    TIME_BETWEEN_RAMP,
+)
+from groundlight import Detector, Groundlight
+from parse_load_test_logs import show_load_test_results
+
+if ENDPOINT_URL == "":
+    raise ValueError("ENDPOINT_URL cannot be an empty string.")
+
+if GROUNDLIGHT_API_TOKEN == "":
+    GROUNDLIGHT_API_TOKEN = os.environ.get("GROUNDLIGHT_API_TOKEN")
+    if GROUNDLIGHT_API_TOKEN is None:
+        raise ValueError("The GROUNDLIGHT_API_TOKEN must be set in config.py or as an environment variable.")
+
+
+def send_image_requests(  # noqa: PLR0913
+    process_id: int, detector: Detector, gl_client: Groundlight, num_requests_per_second: float, duration, log_file: str
+):
+    """Sends image requests to a Groundlight endpoint for a specified duration and logs results."""
+    os.makedirs(os.path.dirname(log_file), exist_ok=True)
+
+    # Prevent errors from appearing in terminal
+    sys.stderr = open(os.devnull, "w")
+
+    start_time = time.time()
+    request_number = 1
+
+    while time.time() - start_time < duration:
+        request_start_time = time.time()
```

Comment:
nit: request_start_time shouldn't include the time to set up `log_data` (its probably pretty insignificant though)